PyInstaller/utils/hooks/gi.py

```diff
@@ -20,27 +20,129 @@
 logger = logging.getLogger(__name__)
 
 
-@isolated.decorate
-def get_gi_libdir(module, version):
-    import os
-    import gi
-    gi.require_version("GIRepository", "2.0")
-    from gi.repository import GIRepository
-    from PyInstaller.depend.bindepend import findSystemLibrary
+class GiModuleInfo:
+    def __init__(self, module, version, hook_api=None):
+        self.name = module
+        self.version = version
+        self.available = False
+        self.sharedlibs = []
+        self.typelib = None
+        self.dependencies = []
+
+        # If hook API is available, use it to override the version from hookconfig.
+        if hook_api is not None:
+            module_versions = get_hook_config(hook_api, 'gi', 'module-versions')
+            if module_versions:
+                version = module_versions.get(module, version)
+
+        logger.debug("Gathering GI module info for %s %s", module, version)
+
+        @isolated.decorate
+        def _get_module_info(module, version):
+            import gi
+            gi.require_version("GIRepository", "2.0")
+            from gi.repository import GIRepository
+
+            repo = GIRepository.Repository.get_default()
+            repo.require(module, version, GIRepository.RepositoryLoadFlags.IREPOSITORY_LOAD_FLAG_LAZY)
+
+            # Shared library/libraries
+            # Comma-separated list of paths to shared libraries, or None if none are associated. Convert to list.
+            sharedlibs = repo.get_shared_library(module)
+            sharedlibs = [lib.strip() for lib in sharedlibs.split(",")] if sharedlibs else []
```

Comment:
This would handle the stripping and the `sharedlibs == ""` case slightly more seamlessly.
```suggestion
            sharedlibs = re.findall(r"[^,\s]+", sharedlibs)
```

Reply:
We still need the `if sharedlibs else []` part, because we must handle the `sharedlibs` being  `None`.

I also think split + strip makes it more obvious what is being done here, although that might just be my relative lack of exposure to regexes :)